### PR TITLE
Add ability to log message content

### PR DESCRIPTION
**Pull request by anurse**
Also improve how browser tests capture logs so you can review them easily from both the command prompt and the browser.

This is for post-preview2 (not worth destabilizing) so it is going in to `dev`.

### CODE REVIEW

File: clients/ts/signalr/src/Transports.ts [hunk 1]
```diff
@@ -320,17 +326,37 @@ export class LongPollingTransport implements ITransport {
     public onclose: TransportClosed;
 }
 
-function getDataDetail(data: any): string {
+function getDataDetail(data: any, includeContent: boolean): string {
     let length: string = null;
     if (data instanceof ArrayBuffer) {
-        length = `Binary data of length ${data.byteLength}`;
+        length = `Binary data of length ${data.byteLength}.`;
+        if (includeContent) {
```
Comment (JamesNK) [inline clients/ts/signalr/src/Transports.ts]: I was thinking yesterday of the same thing 👍

File: clients/ts/signalr/src/Transports.ts [hunk 2]
```diff
@@ -320,17 +326,37 @@ export class LongPollingTransport implements ITransport {
...
 
-async function send(logger: ILogger, transportName: string, httpClient: HttpClient, url: string, accessTokenFactory: () => string, content: string | ArrayBuffer): Promise<void> {
+function formatArrayBuffer(data: ArrayBuffer): string {
+    const view = new Uint8Array(data);
+
+    // Uint8Array.map only supports returning another Uint8Array?
+    let str = "";
+    view.forEach((num) => {
+        const pad = num < 16 ? "0" : "";
+        str += `0x${pad}${num.toString(16)} `;
```
Comment (JamesNK) [inline clients/ts/signalr/src/Transports.ts]: Should we worry about perf for large binary data, e.g. someone sending a 10meg message. This loop will create a lot of strings.

Comment (anurse) [reply clients/ts/signalr/src/Transports.ts]: Yeah, I was thinking that. For right now, that's somewhat mitigated by the fact that it's off by default.

File: clients/ts/FunctionalTests/ts/ConnectionTests.ts [hunk 3]
```diff
@@ -34,33 +39,104 @@ describe("connection", () => {
...
-            const connection = new HttpConnection("echo", {
-                logger: TestLogger.instance,
-                transport: transportType,
+        describe(`over ${TransportType[transportType]}`, () => {
+            it("can send and receive messages", (done) => {
+                const message = "Hello World!";
+                // the url should be resolved relative to the document.location.host
+                // and the leading '/' should be automatically added to the url
+                const connection = new HttpConnection("echo", {
+                    logger: TestLogger.instance,
```
Comment (BrennanConroy) [inline clients/ts/FunctionalTests/ts/ConnectionTests.ts]: pass in `commonOptions`

File: clients/ts/signalr/src/Transports.ts [hunk 4]
```diff
@@ -320,17 +326,37 @@ export class LongPollingTransport implements ITransport {
...
 }
 
-function getDataDetail(data: any): string {
+function getDataDetail(data: any, includeContent: boolean): string {
     let length: string = null;
     if (data instanceof ArrayBuffer) {
-        length = `Binary data of length ${data.byteLength}`;
+        length = `Binary data of length ${data.byteLength}.`;
+        if (includeContent) {
+            length += ` Content: ${formatArrayBuffer(data)}`;
```
Comment (BrennanConroy) [inline clients/ts/signalr/src/Transports.ts]: nit: full stop

File: clients/ts/signalr/src/Transports.ts [hunk 5]
```diff
@@ -320,17 +326,37 @@ export class LongPollingTransport implements ITransport {
...
+        }
     } else if (typeof data === "string") {
-        length = `String data of length ${data.length}`;
+        length = `String data of length ${data.length}.`;
+        if (includeContent) {
+            length += ` Content: ${data}`;
```
Comment (BrennanConroy) [inline clients/ts/signalr/src/Transports.ts]: same

File: SignalR.sln [hunk 6]
```diff
@@ -19,6 +19,9 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 	EndProjectSection
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "samples", "samples", "{C4BC9889-B49F-41B6-806B-F84941B2549B}"
+	ProjectSection(SolutionItems) = preProject
+		clients\ts\FunctionalTests\web.config = clients\ts\FunctionalTests\web.config
```
Comment (davidfowl) [inline SignalR.sln]: huh

**Comment by anurse** [comment]
Yeah, I'll revive this today, thanks for the ping.

File: SignalR.sln (hunk 6 above)
Comment (anurse) [reply SignalR.sln]: 🤷‍♂️

File: clients/ts/signalr/src/Transports.ts (hunk 4 above)
Comment (anurse) [reply clients/ts/signalr/src/Transports.ts]: You moving to New Zealand soon?

**Pull request merged**